src/components/shared/__tests__/Slider.test.tsx

```diff
@@ -1,47 +1,74 @@
-import 'react-native';
-
 import * as React from 'react';
 
+import { fireEvent, render } from '@testing-library/react-native';
+import Rail from '../Slider/Rail';
 import Slider from '../Slider';
-// Note: test renderer must be required after react-native.
-import renderer from 'react-test-renderer';
-
-let props: any;
-let component: React.ReactElement;
-// let testingLib: RenderResult;
-
-const createTestProps = (obj: object): object => ({
-  navigation: {
-    navigate: jest.fn(),
-  },
-  ...obj,
-});
+import { View } from 'react-native';
 
 describe('[Slider] render', () => {
-  beforeEach(() => {
-    props = createTestProps({});
-    component = <Slider {...props} />;
+  it('renders without crashing', () => {
+    const rendered = render(<Slider />).asJSON();
+    expect(rendered).toMatchSnapshot();
+    expect(rendered).toBeTruthy();
   });
+});
+
+describe('[Rail]', () => {
+  const TEST_ID = 'TEST_ID_FOR_MARK';
+  const Mark: React.FC<{ testID: string }> = ({ testID }) => {
+    return <View testID={testID} />;
+  };
 
   it('renders without crashing', () => {
-    const rendered: renderer.ReactTestRendererJSON | null = renderer
-      .create(component)
-      .toJSON();
+    const rendered = render(
+      <Rail mark={<Mark testID={TEST_ID} />} customMarkWidth={1} />,
+    ).asJSON();
     expect(rendered).toMatchSnapshot();
     expect(rendered).toBeTruthy();
   });
 
-  // describe('interactions', () => {
-  //   beforeEach(() => {
-  //     testingLib = render(component);
-  //   });
-
-  //   it('should simulate onClick', () => {
-  //     const btn = testingLib.queryByTestId('btn');
-  //     act(() => {
-  //       fireEvent.press(btn);
-  //     });
-  //     expect(cnt).toBe(3);
-  //   });
-  // });
+  it('calls onMarkPress with correct value and index', () => {
+    const onMarkPress = jest.fn();
+    const STEP = 10;
+    const MARK_COUNT = 5;
+
+    const { getAllByTestId } = render(
+      <Rail
```

Comment:
I agree that rendering component in every single test might be boilerplate, but I'm not exactly sure how extracting a function and specifying props for each test solves the issue.

In my opinion, looking at a test with
```
<Rail
  mark={<Mark testID={TEST_ID} />}
  customMarkWidth={10}
  step={STEP}
  markCount={MARK_COUNT}
  onMarkPress={handleMarkPress}
/>
```
takes less time to understand than a test with
```
renderRail({
  markTestId,
  customMarkWidth,
  step,
  markCount,
  onMarkPress
})
```

I would love to learn if there is a more intuitive way to remove the boilerplate.

Reply:
I agree that your example is a good structure to extract the boilerplate.
The `renderRail` function may have default values inside the function. And It can override the values as your example depends on each test case. :D